ext/AMDGPUExt.jl

```diff
@@ -0,0 +1,20 @@
+module AMDGPUExt
+using KernelAbstractions
+using AMDGPU, ROCKernels, Adapt
+import DiffEqGPU
+
+function DiffEqGPU.EnsembleGPUArray(cpu_offload::Float64)
```

Comment:
we can only really define one of these functions. I only defined it for CUDAExt to not break existing users.

Reply:
@ChrisRackauckas see this comment from Valentin.
we’ll need to define`AMDGPUExt` similar to CUDA, but it may break things as suggested by him

Reply:
No, what he's saying is that we shouldn't have the default dispatch for all backends. It's fine to define this specific one just for CUDA.

Reply:
Yeah make it backwards compatible, but only for CUDA, everyone else will have to select their backend manually.
I have and AMD and a NVIDA GPU in my system and you can't guess at which one is the right one.

Reply:
Sure, I understand now. Should I move the adapt rules to KA then? (Otherwise I’ll have to add AMDGPU as weak deps here)